flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/offset/SeekBinlogByTimestampListener.java

```diff
@@ -0,0 +1,115 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ververica.cdc.connectors.mysql.source.offset;
+
+import com.github.shyiko.mysql.binlog.BinaryLogClient;
+import com.github.shyiko.mysql.binlog.event.Event;
+import com.github.shyiko.mysql.binlog.event.EventHeader;
+import com.github.shyiko.mysql.binlog.event.EventType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** event listener to seek binlog offset by timestamp . */
+public class SeekBinlogByTimestampListener implements BinaryLogClient.EventListener {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SeekBinlogByTimestampListener.class);
+
+    private final long seekTimestamp;
+    private final BinaryLogClient client;
+    private final BinlogOffset maxBinlogOffset;
+    private BinlogOffset binlogOffset;
+    private BinlogOffset lastBinlogOffset;
+
+    public SeekBinlogByTimestampListener(
+            long seekTimestamp, BinaryLogClient client, BinlogOffset maxBinlogOffset) {
+        this.seekTimestamp = seekTimestamp;
+        this.client = client;
+        this.maxBinlogOffset = maxBinlogOffset;
+    }
+
+    @Override
+    public void onEvent(Event event) {
+        try {
+            EventHeader header = event.getHeader();
+            EventType eventType = header.getEventType();
+            long binlogTimestamp = header.getTimestamp();
+
+            // Get the min binlog offset of this binlog file
+            if (binlogOffset == null && eventType == EventType.ROTATE) {
+                binlogOffset = buildBinlogOffset(binlogTimestamp);
+                return;
+            }
+
+            // The first event binlogTimestamp > seekTimestamp , we skip this binlog file directly
+            if (eventType == EventType.FORMAT_DESCRIPTION && seekTimestamp < binlogTimestamp) {
+                LOG.info(
+                        "skip this binlog file {} directly , binlogTimestamp = {}, seekTimestamp = {}",
+                        client.getBinlogFilename(),
+                        binlogTimestamp,
+                        seekTimestamp);
+                client.disconnect();
+            }
+
+            // up to the binlog file max position , exit it
+            if (client.getBinlogPosition() >= maxBinlogOffset.getPosition()
+                    && client.getBinlogFilename().equals(maxBinlogOffset.getFilename())) {
+                LOG.info(
+                        "up to the binlog file max position , exit it , binlogFile = {}, binlogPosition = {}",
+                        client.getBinlogPosition(),
+                        client.getBinlogFilename());
+                client.disconnect();
```

Comment:
Simply disconnecting the binlog client here might cause problem.

e.g., 

1. Current timestamp = 1660530813853, construct maxBinglogOffset accordingly
2. Seek to the binlog file max position, without setting binlogOffset 
3. MysqlBinlogSplit has binlog offset equal to null
4. When MySQL CDC connector assign binlog split, it starts from latest offset with timestsamp = 
1660530924618, ommitting all events between 1660530813853 and 1660530924618